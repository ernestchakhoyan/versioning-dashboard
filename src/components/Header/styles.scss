@import "../../constants/styles/vars";

.header{
  display: flex;
  flex-direction: row;
  width: 100%;
  height: $header_height;
  background-color: $main-theme-color;

  &__logo{
    width: 100px;
    height: 100%;
    justify-content: center;
    align-items: center;
    margin-left: 50px;

    img{
      height: inherit;
      width: auto;
      margin: 0 auto;
    }
  }

  &__title{
    display: flex;
    align-items: center;
    color: white;
  }
}