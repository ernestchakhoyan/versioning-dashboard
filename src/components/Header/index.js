import React from "react";
import Logo from "../../assets/logo.png";
import "./styles.scss"

const Header = (props) => {
	return (
		<header data-test="header-component">
			<div className="wrapper">
				<div className="logo">
					<img
						src={Logo}
						alt="logo"
						data-test="logo-img"
					/>
				</div>
			</div>
		</header>
	)
};

export default Header;