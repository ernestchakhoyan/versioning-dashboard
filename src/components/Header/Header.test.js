import React from "react";
import { shallow } from "enzyme";
import {findBtTestAttr} from "../../utils"
import renderer from "react-test-renderer";
import Header from "./index";


const setUp = (props = {}) => {
	return shallow(<Header {...props} />);
};

describe("Header component", () => {

	let component;

	beforeEach(() => {
		component = setUp();
	});

	it("Should render without errors", () => {
		const wrapper = findBtTestAttr(component,"header-component");
		expect(wrapper.length).toBe(1);
	});

	it("Should render logo", () => {
		const logo = findBtTestAttr(component,"logo-img");
		expect(logo.length).toBe(1);
	});

	it("Capture component snapshot" ,() => {
		const tree = renderer.create(<Header />).toJSON();

		expect(tree).toMatchSnapshot();
	})
});