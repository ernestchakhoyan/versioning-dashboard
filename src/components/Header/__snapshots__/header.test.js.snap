// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Capture component snapshot 1`] = `
<header
  className="header"
  data-test="header-component"
>
  <div
    className="header__logo"
  >
    <img
      alt="logo"
      data-test="logo-img"
      src="logo.png"
    />
  </div>
  <div
    className="header__title"
  >
    Behive Versioning Dashboard
  </div>
</header>
`;
